src/engine/BMInterface.php

```diff
@@ -1074,7 +1050,7 @@ protected function resolve_random_button_selection(BMGame &$game) {
             $randIdx = bm_rand(0, $nButtons - 1);
             $buttonId = $allButtonData[$randIdx]['buttonId'];
 
-            $this->choose_button($game, $buttonId, $buttonIdx);
+            $this->choose_button($game, $buttonId, $playerIdx);
```

Comment:
Pass `$player` instead of `$playerIdx`, and have the database insert use `$player->position`.  Then you don't need to reference `$playerIdx` here.